cloud_functions/climateiq_data_export_cf/main_test.py

```diff
@@ -334,27 +342,217 @@ def test_export_model_predictions_h3_centroids_within_chunk(
         "col_count": 3,
         "chunks": {
             "chunk-id": {
+                "id": "chunk-id",
                 "row_count": 2,
                 "col_count": 3,
                 "x_ll_corner": 500,
                 "y_ll_corner": 100,
-                "x_index": 0,
-                "y_index": 0,
+                "x_index": 1,
+                "y_index": 1,
+            }
+        },
+    }
+    mock_firestore_client().collection().document().get().to_dict.return_value = (
+        metadata
+    )
+
+    # Neighbor chunks do not exist.
+    mock_firestore_client().collection().document().collection().where().where(). \
+        limit().get().exists = (
+            False
+    )
+
+    with pytest.raises(ValueError) as exc_info:
+        main.export_model_predictions(event)
+
+    assert "Neighbor chunk at index" in str(
+        exc_info.value
+    ) and "is missing from the study area" in str(exc_info.value)
+
+
+@mock.patch.object(storage, "Client", autospec=True)
+@mock.patch.object(firestore, "Client", autospec=True)
+def test_export_model_predictions_invalid_neighbor_chunk(
+    mock_firestore_client, mock_storage_client
+) -> None:
+    attributes = {
+        "type": "google.cloud.storage.object.v1.finalized",
+        "source": "source",
+    }
+    data = {
+        "bucket": "climateiq-predictions",
+        "name": "prediction-type/model-id/study-area-name/scenario-id/chunk-id",
+    }
+    event = http.CloudEvent(attributes, data)
+
+    # Build mock Storage object
+    predictions = '{"instance": [1], "prediction": [[1, 2, 3], [4, 5, 6]]}\n'
+    with mock_storage_client().bucket("").blob("").open() as mock_fd:
+        mock_fd.__iter__.return_value = iter(predictions.splitlines())
+
+    # Build mock Firestore document
+    metadata = {
+        "name": "study_area_name",
+        "cell_size": 10,
+        "crs": "EPSG:32618",
+        "row_count": 2,
+        "col_count": 3,
+        "chunks": {
+            "chunk-id": {
+                "id": "chunk-id",
+                "row_count": 3,
+                "col_count": 2,
+                "x_ll_corner": 500,
+                "y_ll_corner": 100,
+                "x_index": 1,
+                "y_index": 1,
+            }
+        },
+    }
+    mock_firestore_client().collection().document().get().to_dict.return_value = (
+        metadata
+    )
+
+    neighbor_metadata = metadata["chunks"]["chunk-id"].copy()
+    neighbor_metadata.pop("row_count")
+    mock_firestore_client().collection().document().collection().where().where(). \
+        limit().get().to_dict.return_value = (
+            neighbor_metadata
+    )
+
+    with pytest.raises(ValueError) as exc_info:
+        main.export_model_predictions(event)
+
+    assert (
```

Comment:
Instead of this you can use the `match` param in `pytest.raises`: https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.raises

Reply:
I can actually just hard code the index here